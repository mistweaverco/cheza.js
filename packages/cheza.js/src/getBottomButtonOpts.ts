import type { ChezaDataStore } from '../types'
import type { ChezaButtonOpts } from './createButton'
import { createSVG } from './createSVG'

export const getBottomButtonOpts = (dataStore: ChezaDataStore): ChezaButtonOpts[] => {
  return [
    {
      name: 'play',
      ariaLabel: 'Play',
      title: 'Play',
      svg: createSVG('play', {}),
      on: [
        {
          name: 'click',
          callback: () => {
            void dataStore.videoElement.play()
          }
        }
      ]
    },
    {
      name: 'pause',
      ariaLabel: 'Pause',
      title: 'Pause',
      svg: createSVG('pause', {}),
      hidden: true,
      on: [
        {
          name: 'click',
          callback: () => {
            dataStore.videoElement.pause()
          }
        }
      ]
    },
    {
      name: 'volume',
      ariaLabel: 'Mute',
      title: 'Mute',
      svg: createSVG('volume', {}),
      on: [
        {
          name: 'click',
          callback: () => {
            dataStore.videoElement.volume = 0
          }
        }
      ]
    },
    {
      name: 'muted',
      ariaLabel: 'Unmute',
      title: 'Unmute',
      svg: createSVG('muted', {}),
      hidden: true,
      on: [
        {
          name: 'click',
          callback: () => {
            dataStore.videoElement.volume = 1
          }
        }
      ]
    },
    {
      name: 'enter-fullscreen',
      ariaLabel: 'Enter Fullscreen',
      title: 'Enter Fullscreen',
      svg: createSVG('enter-fullscreen', {}),
      on: [
        {
          name: 'click',
          callback: () => {
            void dataStore.rootElement.requestFullscreen()
          }
        }
      ]
    },
    {
      name: 'exit-fullscreen',
      ariaLabel: 'Exit Fullscreen',
      title: 'Exit Fullscreen',
      svg: createSVG('exit-fullscreen', {}),
      hidden: true,
      on: [
        {
          name: 'click',
          callback: () => {
            // TODO make this work
          }
        }
      ]
    }
  ]
}
