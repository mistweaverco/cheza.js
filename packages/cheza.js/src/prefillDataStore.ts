import type { ChezaDataStore } from '../types'

export const prefillDataStore = (videoElement: HTMLVideoElement): ChezaDataStore => {
  const rootElement = document.createElement('div')
  const progressContainer = document.createElement('div')
  const progress = document.createElement('div')
  const controlsTop = document.createElement('div')
  const controlsBottom = document.createElement('div')
  const videoContainer = document.createElement('div')
  const loadingSpinner = document.createElement('div')
  const activePlugins = new Map()
  return {
    rootElement,
    videoElement,
    progressContainer,
    progress,
    controlsTop,
    controlsBottom,
    videoContainer,
    loadingSpinner,
    activePlugins
  }
}
