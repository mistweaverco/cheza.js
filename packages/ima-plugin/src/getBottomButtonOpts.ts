import type { ChezaIMADataStore, google } from '../types'
import type { ChezaButtonOpts } from './createButton'
import { createSVG } from './createSVG'
import { enterFullscreen, exitFullscreen } from './fullscreenUtils'

interface AllButtonOpts {
  left: ChezaButtonOpts[]
  right: ChezaButtonOpts[]
}

export enum GetBottomButtonOptsDirection {
  LEFT = 'left',
  RIGHT = 'right'
}

export const getBottomButtonOpts = (dataStore: ChezaIMADataStore, direction: GetBottomButtonOptsDirection): ChezaButtonOpts[] => {
  const all: AllButtonOpts = {
    left: [],
    right: []
  }
  all.left = [
    {
      name: 'play',
      ariaLabel: 'Play',
      title: 'Play',
      svg: createSVG('play'),
      on: [
        {
          name: 'click',
          callback: () => {
            const am = dataStore.adsManager as google.ima.AdsManager
            am.resume()
          }
        }
      ]
    },
    {
      name: 'pause',
      ariaLabel: 'Pause',
      title: 'Pause',
      svg: createSVG('pause'),
      hidden: true,
      on: [
        {
          name: 'click',
          callback: () => {
            const am = dataStore.adsManager as google.ima.AdsManager
            am.pause()
          }
        }
      ]
    },
    {
      name: 'volume',
      ariaLabel: 'Mute',
      title: 'Mute',
      svg: createSVG('volume'),
      on: [
        {
          name: 'click',
          callback: () => {
            const am = dataStore.adsManager as google.ima.AdsManager
            am.setVolume(0)
          }
        }
      ]
    },
    {
      name: 'muted',
      ariaLabel: 'Unmute',
      title: 'Unmute',
      svg: createSVG('muted'),
      hidden: true,
      on: [
        {
          name: 'click',
          callback: () => {
            const am = dataStore.adsManager as google.ima.AdsManager
            am.setVolume(1)
          }
        }
      ]
    }
  ]
  all.right = [
    {
      name: 'enter-fullscreen',
      ariaLabel: 'Enter Fullscreen',
      title: 'Enter Fullscreen',
      svg: createSVG('enter-fullscreen'),
      on: [
        {
          name: 'click',
          callback: () => {
            enterFullscreen(dataStore.rootElement as HTMLElement, dataStore.imaVideoElement as HTMLVideoElement)
          }
        }
      ]
    },
    {
      name: 'exit-fullscreen',
      ariaLabel: 'Exit Fullscreen',
      title: 'Exit Fullscreen',
      svg: createSVG('exit-fullscreen'),
      hidden: true,
      on: [
        {
          name: 'click',
          callback: () => {
            exitFullscreen(dataStore.imaVideoElement as HTMLVideoElement)
          }
        }
      ]
    }
  ]
  return all[direction]
}
