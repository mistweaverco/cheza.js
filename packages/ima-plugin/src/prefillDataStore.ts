import type { ChezaIMADataStore } from '../types'

export const prefillDataStore = (): ChezaIMADataStore => {
  const root = document.createElement('div')
  const adContainer = document.createElement('div')
  const controlsTop = document.createElement('div')
  const controlsBottom = document.createElement('div')
  const loadingSpinner = document.createElement('div')
  return {
    defaultUIElement: undefined,
    videoElement: undefined,
    imaVideoElement: undefined,
    totalAds: 0,
    adPosition: 0,
    adsManager: undefined,
    adsLoader: undefined,
    adDisplayContainer: undefined,
    rootElement: undefined,
    adsInitialized: false,
    google: undefined,
    adsLoaded: false,
    opts: {},
    ui: {
      root,
      overlay: document.createElement('div'),
      adContainer,
      controlsTop,
      controlsBottom,
      loadingSpinner
    }
  }
}
