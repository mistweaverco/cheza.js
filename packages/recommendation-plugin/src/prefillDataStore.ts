import type { ChezaRecommendationDataStore } from '../types'

export const prefillDataStore = (): ChezaRecommendationDataStore => {
  const root = document.createElement('div')
  const content = document.createElement('div')
  return {
    defaultUIElement: undefined,
    videoElement: undefined,
    rootElement: undefined,
    intervalTicker: null,
    opts: {},
    ui: {
      root,
      content
    }
  }
}
