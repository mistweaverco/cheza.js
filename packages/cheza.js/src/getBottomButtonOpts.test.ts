import { prefillDataStore } from './prefillDataStore'
import { getBottomButtonOpts } from './getBottomButtonOpts'

describe('getBottomButtonOpts', () => {
  const video = document.createElement('video')
  const dataStore = prefillDataStore(video)
  it('should match the snapshot', () => {
    expect(getBottomButtonOpts(dataStore)).toMatchSnapshot()
  })
})
