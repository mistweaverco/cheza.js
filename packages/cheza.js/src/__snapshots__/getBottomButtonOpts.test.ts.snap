// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBottomButtonOpts should match the snapshot 1`] = `
[
  {
    "ariaLabel": "Play",
    "name": "play",
    "on": [
      {
        "callback": [Function],
        "name": "click",
      },
    ],
    "svg": <svg
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#cheza-play"
      />
    </svg>,
    "title": "Play",
  },
  {
    "ariaLabel": "Pause",
    "hidden": true,
    "name": "pause",
    "on": [
      {
        "callback": [Function],
        "name": "click",
      },
    ],
    "svg": <svg
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#cheza-pause"
      />
    </svg>,
    "title": "Pause",
  },
  {
    "ariaLabel": "Mute",
    "name": "volume",
    "on": [
      {
        "callback": [Function],
        "name": "click",
      },
    ],
    "svg": <svg
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#cheza-volume"
      />
    </svg>,
    "title": "Mute",
  },
  {
    "ariaLabel": "Unmute",
    "hidden": true,
    "name": "muted",
    "on": [
      {
        "callback": [Function],
        "name": "click",
      },
    ],
    "svg": <svg
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#cheza-muted"
      />
    </svg>,
    "title": "Unmute",
  },
  {
    "ariaLabel": "Enter Fullscreen",
    "name": "enter-fullscreen",
    "on": [
      {
        "callback": [Function],
        "name": "click",
      },
    ],
    "svg": <svg
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#cheza-enter-fullscreen"
      />
    </svg>,
    "title": "Enter Fullscreen",
  },
  {
    "ariaLabel": "Exit Fullscreen",
    "hidden": true,
    "name": "exit-fullscreen",
    "on": [
      {
        "callback": [Function],
        "name": "click",
      },
    ],
    "svg": <svg
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#cheza-exit-fullscreen"
      />
    </svg>,
    "title": "Exit Fullscreen",
  },
]
`;
